components/TaskContainer.vue

```diff
@@ -92,16 +114,24 @@ export default {
   },
   mounted() {
     this.$store.dispatch('todos/setTotalPage');
-    this.$store.dispatch('todos/filterTaskList', this.filterOptions[0]);
+    this.$store.dispatch('todos/setActiveFilterOption', this.filterOptions[0]);
+    this.$store.dispatch('todos/filterTaskList');
   },
   methods: {
-    showAddTodoCard() {
+    async showAddTodoCard() {
+      await this.$store.dispatch('todos/setSearchText', '');
+      this.$store.dispatch('todos/setShowSearchField', false);
       this.showAddCard = true;
     },
     checkForm(e) {
       e.preventDefault();
+      // Sanitize the user input by removing any HTML tags
```

Comment:
we can move this fuction `checkForm` to a common place as this validation function is used elsewhere as well.

Reply:
Resolved